File: 118-pascals-triangle/pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    let num = [];
    let n = numRows;
    for(let i=0;i<n;i++){
        
        let nums=[];
        for(let j=0;j<i+1;j++){
            
            if(i===j || j===0) nums.push(1);
            else nums.push(num[i-1][j-1]+num[i-1][j]);
            
        }
        num.push(nums);
    }
    return num;
};

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for(int i=0;i<numRows;i++){
            ArrayList<Integer> arr = new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0 || i==j) arr.add(1);
                else arr.add(res.get(i-1).get(j-1)+res.get(i-1).get(j));
            }
            res.add(arr);
        }
        return res;
    }
}